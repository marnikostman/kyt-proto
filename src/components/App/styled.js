import { css } from 'pretty-lights';
import logo from '../../images/icon-kytLogo_large-252x252.svg';
import { contentWidth, padding } from '../../shared-styles/variables';

export const headerClass = css`
  text-align: center;
`;

export const navClass = css`
  width: ${contentWidth}px;
  margin: 0 auto;
  padding: ${padding}px;
  text-align: center;
`;

export const contentClass = css`
  width: ${contentWidth}px;
  margin: ${padding}px auto;
`;

export const navItemClass = css`
  display: inline-block;
`;

export const linkClass = css`
  padding: ${padding}px;
  color: #00a68f;
  font-size: 18px;
`;

export const logoClass = css`
  display: block;
  width: 252px;
  height: 252px;
  margin: ${padding}px auto;
  background: url(${logo});
`;
