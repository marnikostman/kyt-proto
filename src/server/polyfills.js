// kyt dependency for polyfills, using babel-preset-env.
