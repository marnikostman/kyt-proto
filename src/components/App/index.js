import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import dynamic from 'kyt-runtime/dynamic';
import * as styles from './styled';

export const Home = dynamic(() => import(/* webpackChunkName: "home" */ '../Home'));
export const Tools = dynamic(() => import(/* webpackChunkName: "tools" */ '../Tools'));

function App() {
  return (
    <div>
      <i className={styles.logoClass} />
      <ul className={styles.navClass}>
        <li className={styles.navItemClass}>
          <Link className={styles.linkClass} to="/">
            Home
          </Link>
        </li>
        <li className={styles.navItemClass}>
          <Link className={styles.linkClass} to="/tools">
            Tools
          </Link>
        </li>
      </ul>
      <div className={styles.contentClass}>
        <Switch>
          <Route path="/" exact component={Home} />
          <Route path="/tools" component={Tools} />
        </Switch>
      </div>
    </div>
  );
}

export default App;
