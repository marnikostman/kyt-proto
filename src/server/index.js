import express from 'express';
import compression from 'compression';
import path from 'path';
import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { preloadDynamicImports, DynamicImports, getBundles } from 'kyt-runtime/server';
import { extractCritical } from 'pretty-lights/server';
import template from './template';
import App from '../components/App';

const port = parseInt(KYT.SERVER_PORT, 10);
const app = express();

// Remove annoying Express header addition.
app.disable('x-powered-by');

// Compress (gzip) assets in production.
app.use(compression());

// Setup the public directory so that we can server static assets.
app.use(express.static(path.join(process.cwd(), KYT.PUBLIC_DIR)));

// Setup server side routing.
app.get('*', (req, res) => {
  const context = {};
  const modules = [];

  const { html, ids, css } = extractCritical(
    renderToString(
      <DynamicImports report={moduleName => modules.push(moduleName)}>
        <StaticRouter location={req.url} context={context}>
          <App />
        </StaticRouter>
      </DynamicImports>
    )
  );

  if (context.status && context.status.code) {
    let message = context.status.message || 'Error!';
    if (!context.status.message && Number(context.status.code) === 404) {
      message = 'Not found';
    }
    res.status(context.status.code).send(message);
  } else if (context.redirect && context.redirect.url) {
    res.redirect(context.redirect.code || 302, context.redirect.url);
  }

  res.status(200).send(
    template({
      html,
      ids,
      css,
      bundles: getBundles({ modules }),
    })
  );
});

preloadDynamicImports().then(() => {
  app.listen(port, () => {
    console.log(`âœ…  server started on port: ${port}`); // eslint-disable-line no-console
  });
});
