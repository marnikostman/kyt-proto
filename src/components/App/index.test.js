import React from 'react';
import { shallow, mount } from 'enzyme';
import { StaticRouter } from 'react-router-dom';
import { DynamicImports } from 'kyt-runtime/server';
import App, { Home, Tools } from '.';

it('Test example', () => {
  const wrapper = shallow(<App>test</App>);

  expect(wrapper.is('div')).toBeTruthy();
});

it('Test home route', async () => {
  await Home.preload();
  const modules = [];
  const wrapper = mount(
    <DynamicImports report={module => modules.push(module)}>
      <StaticRouter location="/" context={{}}>
        <App />
      </StaticRouter>
    </DynamicImports>
  );

  expect(wrapper).toMatchSnapshot();
  expect(modules).toMatchSnapshot();
});

it('Test tools route', async () => {
  await Tools.preload();
  const modules = [];
  const wrapper = mount(
    <DynamicImports report={module => modules.push(module)}>
      <StaticRouter location="/tools" context={{}}>
        <App />
      </StaticRouter>
    </DynamicImports>
  );

  expect(wrapper).toMatchSnapshot();
  expect(modules).toMatchSnapshot();
});
