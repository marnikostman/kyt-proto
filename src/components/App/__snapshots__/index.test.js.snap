// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test home route 1`] = `
.lights-9 {
  width: 500px;
  margin: 0 auto;
  padding: 15px;
  text-align: center;
}

.lights-12 {
  width: 500px;
  margin: 15px auto;
}

.lights-4 {
  display: inline-block;
}

.lights-1 {
  padding: 15px;
  color: #00a68f;
  font-size: 18px;
}

.lights-0 {
  display: block;
  width: 252px;
  height: 252px;
  margin: 15px auto;
  background: url(file-stub);
}

.lights-10 + .lights-10 {
  padding-top: 15px;
}

<Capture
  report={[Function]}
>
  <StaticRouter
    context={Object {}}
    location="/"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      staticContext={Object {}}
    >
      <App>
        <div>
          <i
            className="lights-0"
          />
          <ul
            className="lights-9"
          >
            <li
              className="lights-4"
            >
              <Link
                className="lights-1"
                to="/"
              >
                <LinkAnchor
                  className="lights-1"
                  href="/"
                  navigate={[Function]}
                >
                  <a
                    className="lights-1"
                    href="/"
                    onClick={[Function]}
                  >
                    Home
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li
              className="lights-4"
            >
              <Link
                className="lights-1"
                to="/tools"
              >
                <LinkAnchor
                  className="lights-1"
                  href="/tools"
                  navigate={[Function]}
                >
                  <a
                    className="lights-1"
                    href="/tools"
                    onClick={[Function]}
                  >
                    Tools
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </ul>
          <div
            className="lights-12"
          >
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                exact={true}
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/"
              >
                <LoadableComponent
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  staticContext={Object {}}
                >
                  <Home
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    staticContext={Object {}}
                  >
                    <section>
                      <p
                        className="lights-10"
                      >
                        Welcome to the 
                        <strong>
                          Universal React Starter-kyt
                        </strong>
                        . This starter kyt should serve as the base for an advanced, server-rendered React app.
                      </p>
                      <p
                        className="lights-10"
                      >
                        Check out the Tools section for an outline of the libraries that are used in this Starter-kyt.
                      </p>
                    </section>
                  </Home>
                </LoadableComponent>
              </Route>
            </Switch>
          </div>
        </div>
      </App>
    </Router>
  </StaticRouter>
</Capture>
`;

exports[`Test home route 2`] = `
Array [
  "../Home",
]
`;

exports[`Test tools route 1`] = `
.lights-9 {
  width: 500px;
  margin: 0 auto;
  padding: 15px;
  text-align: center;
}

.lights-15 {
  width: 500px;
  margin: 15px auto;
}

.lights-4 {
  display: inline-block;
}

.lights-1 {
  padding: 15px;
  color: #00a68f;
  font-size: 18px;
}

.lights-0 {
  display: block;
  width: 252px;
  height: 252px;
  margin: 15px auto;
  background: url(file-stub);
}

.lights-10 + .lights-10 {
  padding-top: 5px;
}

<Capture
  report={[Function]}
>
  <StaticRouter
    context={Object {}}
    location="/tools"
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/tools",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      staticContext={Object {}}
    >
      <App>
        <div>
          <i
            className="lights-0"
          />
          <ul
            className="lights-9"
          >
            <li
              className="lights-4"
            >
              <Link
                className="lights-1"
                to="/"
              >
                <LinkAnchor
                  className="lights-1"
                  href="/"
                  navigate={[Function]}
                >
                  <a
                    className="lights-1"
                    href="/"
                    onClick={[Function]}
                  >
                    Home
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li
              className="lights-4"
            >
              <Link
                className="lights-1"
                to="/tools"
              >
                <LinkAnchor
                  className="lights-1"
                  href="/tools"
                  navigate={[Function]}
                >
                  <a
                    className="lights-1"
                    href="/tools"
                    onClick={[Function]}
                  >
                    Tools
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </ul>
          <div
            className="lights-15"
          >
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/tools",
                    "url": "/tools",
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/tools",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/tools"
              >
                <LoadableComponent
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/tools",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/tools",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/tools",
                      "url": "/tools",
                    }
                  }
                  staticContext={Object {}}
                >
                  <Tools
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/tools",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/tools",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/tools",
                        "url": "/tools",
                      }
                    }
                    staticContext={Object {}}
                  >
                    <ul>
                      <li
                        className="lights-10"
                      >
                        <a
                          href="https://expressjs.com/"
                        >
                          Express
                        </a>
                         - server-side rendering
                      </li>
                      <li
                        className="lights-10"
                      >
                        <a
                          href="https://reactjs.org/"
                        >
                          React
                        </a>
                         - component library
                      </li>
                      <li
                        className="lights-10"
                      >
                        <a
                          href="https://reacttraining.com/react-router/"
                        >
                          React Router
                        </a>
                         - server and browser routing
                      </li>
                      <li
                        className="lights-10"
                      >
                        <a
                          href="https://github.com/nytimes/pretty-lights"
                        >
                          Pretty Lights
                        </a>
                         - CSS-in-JS with a reliable API
                      </li>
                      <li
                        className="lights-10"
                      >
                        <a
                          href="https://airbnb.io/enzyme/"
                        >
                          Enzyme
                        </a>
                         - React component testing
                      </li>
                    </ul>
                  </Tools>
                </LoadableComponent>
              </Route>
            </Switch>
          </div>
        </div>
      </App>
    </Router>
  </StaticRouter>
</Capture>
`;

exports[`Test tools route 2`] = `
Array [
  "../Tools",
]
`;
